--- a/modules/avcodec/encode.c
+++ b/modules/avcodec/encode.c
@@ -106,6 +106,6 @@
 	if (!encoder) {
-		warning("avcodec: %s: could not init encoder\n",
-			avcodec_get_name(st->codec_id));
-		err = ENOENT;
-		goto out;
+		warning("avcodec: %s: bypass encoder for passthrough\n",
+			avcodec_get_name(st->codec_id));
+		ves->ctx = NULL;
+		goto skip_context;
 	}
@@ -119,2 +119,3 @@
+skip_context:
 	if (ves->ctx) {
 		ves->ctx->bit_rate = prm->bitrate;
@@ -241,2 +242,4 @@
 	if (!ves || !frame)
 		return EINVAL;
+	if (!ves->ctx)
+		return 0;
