--- a/modules/avcodec/encode.c
+++ b/modules/avcodec/encode.c
@@ -103,19 +103,20 @@
 	}
 
 	encoder = avcodec_find_encoder(st->codec_id);
 	if (!encoder) {
-		warning("avcodec: %s: could not init encoder\n",
-			avcodec_get_name(st->codec_id));
-		err = ENOENT;
-		goto out;
+		warning("avcodec: %s: bypass encoder for passthrough\n",
+			avcodec_get_name(st->codec_id));
+		ves->ctx = NULL;
+		goto skip_context;
 	}
 
 	ves->ctx = avcodec_alloc_context3(encoder);
 	if (!ves->ctx) {
 		err = ENOMEM;
 		goto out;
 	}
 
+skip_context:
 	if (ves->ctx) {
 		ves->ctx->bit_rate = prm->bitrate;
 		ves->ctx->width    = prm->width;
@@ -239,6 +240,9 @@
 	int err = 0;
 
 	if (!ves || !frame)
 		return EINVAL;
+
+	if (!ves->ctx)
+		return 0;
 
 	if (update) {
